File: AAS.Tools/Types/Results/Result.cs
using System.Text.Json.Serialization;

namespace AAS.Tools.Types.Results;

public class Result
{
    public Error[] Errors { get; }

    public bool IsSuccess => Errors.Length == 0;

    [JsonConstructor]
    public Result(List<Error> errors)
    {
        Errors = errors.ToArray();
    }

    public static Result Success()
    {
        return new Result(new List<Error>());
    }

    public static Result Fail(Error error)
    {

        return new Result(new List<Error>() { error });
    }

    public static Result Fail(string error)
    {

        return new Result(new List<Error>() { new Error(error) });
    }
}

File: AAS.BackOffice/client_app/src/tools/types/results/result.ts
import { ErrorData, mapToErrorData } from './errorData';

export class Result {
    public isSuccess = this.errors.length === 0;

    constructor(
        public errors: ErrorData[]
    ) { }

    public static success<T>(): Result {
        return new Result([]);
    }

    public static fail(errors: ErrorData[]): Result {
        return new Result(errors);
    }

    public getErrorsString = (): string => {
        return this.errors.map(error => error.message).join('. ')
    }
}

export const mapToResult = (result: any): Result => {
    if (!result.isSuccess)
        return Result.fail(mapToErrorData(result.errors))
    return Result.success();
}
